tests/phpunit/AutoReview/Event/SubscriberTest.php

```diff
@@ -0,0 +1,185 @@
+
+        foreach ($subscriberMethods as $method) {
+            if (!$this->isSubscriptionMethod($method)) {
+                continue;
+            }
+
+            $this->assertIsSubscriptionMethod($subscriberClass, $method);
+        }
+    }
+
+    /**
+     * @dataProvider Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberClassesProvider
+     *
+     * @param class-string $subscriberClass
+     */
+    public function test_the_subscribed_methods_can_all_be_collected(string $subscriberClass): void
+    {
+        /** @var EventSubscriber $subscriber */
+        $subscriber = (new ReflectionClass($subscriberClass))->newInstanceWithoutConstructor();
+
+        $subscriber->getSubscribedEvents();
+
+        $this->addToAssertionCount(1);
+    }
+
+    /**
+     * @dataProvider Infection\Tests\AutoReview\Event\SubscriberProvider::subscriberSubscriptionMethodsProvider
+     *
+     * @param class-string $subscriberClass
+     */
+    public function test_all_subscription_methods_are_listed_by_the_subscriber(
+        string $subscriberClass,
+        array $subscriptionMethodNames
+    ): void {
+        $subscriptionMethodNames = array_flip($subscriptionMethodNames);
+
+        $subscriberMethods = (new ReflectionClass($subscriberClass))->getMethods();
+
+        foreach ($subscriberMethods as $method) {
+            if (!$this->isSubscriptionMethod($method)) {
+                continue;
+            }
+
+            $subscriptionName = $method->getName();
+
+            $this->assertArrayHasKey(
+                $subscriptionName,
+                $subscriptionMethodNames,
+                sprintf(
+                    'Expected the subscription method "%s" to be listed in "%s::%s()"',
+                    $subscriptionName,
+                    $subscriberClass,
+                    'getSubscribedEvents'
+                )
+            );
+
+            unset($subscriptionMethodNames[$subscriptionName]);
+        }
+
+        $this->assertCount(
+            0,
+            $subscriptionMethodNames,
+            sprintf(
+                'Expected to find the following subscription method(s) on "%s": "%s". %s'
+                . 'Either add them or remove them from "%s::%s()"',
+                $subscriberClass,
+                implode('","', array_keys($subscriptionMethodNames)),
+                PHP_EOL,
+                $subscriberClass,
+                'getSubscribedEvents'
+            )
+        );
+    }
+
+    private function isSubscriptionMethod(ReflectionMethod $method): bool
+    {
+        return !$method->isConstructor()
+            && $method->isPublic()
+            && $method->getName() !== 'getSubscribedEvents'
+        ;
+    }
+
+    /**
+     * @param class-string $subscriberClass
+     */
+    private function assertIsSubscriptionMethod(string $subscriberClass, ReflectionMethod $method): void
+    {
+        $this->assertSame(
```

Comment:
Doesn't matter much, but perhaps using `assertCount()` here makes a bit more sense?

```diff
diff --git a/tests/phpunit/AutoReview/Event/SubscriberTest.php b/tests/phpunit/AutoReview/Event/SubscriberTest.php
index e38d284e..db789702 100644
--- a/tests/phpunit/AutoReview/Event/SubscriberTest.php
+++ b/tests/phpunit/AutoReview/Event/SubscriberTest.php
@@ -36,6 +36,7 @@ declare(strict_types=1);
 namespace Infection\Tests\AutoReview\Event;

 use function array_keys;
+use function count;
 use function implode;
 use Infection\Event\Subscriber\EventSubscriber;
 use const PHP_EOL;
@@ -143,15 +144,15 @@ final class SubscriberTest extends TestCase
      */
     private function assertIsSubscriptionMethod(string $subscriberClass, ReflectionMethod $method): void
     {
-        $this->assertSame(
+        $this->assertCount(
             1,
-            $method->getNumberOfParameters(),
+            $method->getParameters(),
             sprintf(
                 'Expected a subscription method "%s::%s" to have exactly one parameter: the'
                 . ' event. Got "%d"',
                 $subscriberClass,
                 $method->getName(),
-                $method->getNumberOfParameters()
+                count($method->getNumberOfParameters())
             )
         );

```

Reply:
Oups thought you were done with the review

I think it's ok: what matters here is the error message. I don't have a strong preference but one is slightly more verbose than the other so might go for the shorter version